docs/content/error/$http/baddata.ngdoc

```diff
@@ -0,0 +1,13 @@
+@ngdoc error
+@name $http:baddata
+@fullName Bad JSON Data
+@description
+
+This error occurs when the data parameter passed to the {@link ng.$http `defaultHttpResponseTransform`} service is not a valid JSON object.
+`defaultHttpResponseTransform` expects the first parameter as a valid JSON object if the second parameter of headers specifies a Content-Type of JSON.
```

Comment:
Don't mention `defaultHttpResponseTransform`, because it is private and the users don't know what it is.
Just say that _"The default `transformResponse` will try to parse the response as JSON if the `Content-Type` header is `application/json` or the response looks like a valid JSON-stringified object or array"_.